tests/test_validators.py

```diff
@@ -1345,3 +1346,33 @@ class Model(BaseModel):
         {'loc': ('foo',), 'msg': 'the list has duplicated items', 'type': 'value_error.list.unique_items'},
         {'loc': ('bar',), 'msg': 'the list has duplicated items', 'type': 'value_error.list.unique_items'},
     ]
+
+
+@pytest.mark.parametrize(
+    'func,allow_reuse',
+    [
+        pytest.param(partial, False, id='`partial` and check for reuse'),
+        pytest.param(partial, True, id='`partial` and ignore reuse'),
+        pytest.param(partialmethod, False, id='`partialmethod` and check for reuse'),
+        pytest.param(partialmethod, False, id='`partialmethod` and ignore reuse'),
```

Comment:
```suggestion
        pytest.param(partialmethod, True, id='`partialmethod` and ignore reuse'),
```
???

Reply:
And that's the reason I used a generator at first: I'm really bad at copy paste

Reply:
I think this is much clearer, although there's a few "copied" lines.

For future reference, if you do want to do the "product" thing, imho you'd be better off using two decorators, one for each argument rather than the `product` function.